tools/firtool/firtool.cpp

```diff
@@ -438,6 +439,37 @@ static LogicalResult printOp(Operation *op, raw_ostream &os) {
   return success();
 }
 
+// If requested, finalize and print the MLIR into mlirOutFile. The final MLIR
+// will erase any sv.verbatim ops that represented contents for sideband files,
+// so this should always be called after ExportVerilog.
+static LogicalResult finalizeAndPrintMlir(ModuleOp moduleOp) {
+  if (mlirOutFile.empty())
+    return success();
+
+  std::string mlirOutError;
+  auto mlirFile = openOutputFile(mlirOutFile, &mlirOutError);
+  if (!mlirFile) {
+    llvm::errs() << mlirOutError;
+    return failure();
+  }
+
+  // Finalize the MLIR by erasing any sv.verbatim ops for sideband files.
+  for (auto verbatim : llvm::make_early_inc_range(
```

Comment:
Yeah this is a fair point. I don't really want to add this functionality to ExportVerilog, even behind a flag, because I'm hoping we can continue moving away from doing any mutation during that pass. I'll make a little helper function or nanopass to implement this functionality.

Reply:
I just added a little pass in [dab6c4a](https://github.com/llvm/circt/pull/5181/commits/dab6c4afbf8a38c3735532a0f68ac040904cb180), let me know what you think.